contracts/SlashIndicator.sol

```diff
@@ -161,6 +185,93 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
     emit indicatorCleaned();
   }
 
+  function submitFinalityViolationEvidence(FinalityEvidence calldata _evidence) external onlyInit onlyRelayer {
+    if (finalitySlashRewardRatio == 0) {
+      finalitySlashRewardRatio = INIT_FINALITY_SLASH_REWARD_RATIO;
+    }
+    uint256 srcNumA = _evidence.voteA.srcNum;
+    uint256 tarNumA = _evidence.voteA.tarNum;
+    uint256 srcNumB = _evidence.voteB.srcNum;
+    uint256 tarNumB = _evidence.voteB.tarNum;
+
+    require(!(_evidence.voteA.srcHash == _evidence.voteB.srcHash &&
+    _evidence.voteA.tarHash == _evidence.voteB.tarHash), "two identical votes");
+    require(srcNumA < tarNumA && srcNumB < tarNumB, "source number bigger than target number");
+
+    // Vote rules check
+    if (!((srcNumA < srcNumB && srcNumB < tarNumB && tarNumB < tarNumA) ||
```

Comment:
`srcNumB < tarNumB` and `srcNumA < tarNumA` already  been checked, is't it?

Reply:
`logical operation` seems not correct here,    `statementA || statement B && statementC` means `statementA || ( statement B && statementC)`, it is not what you expect, is it?

Reply:
Fixed